### PR TITLE
Emails can be returned by id from inbox section

### PR DESCRIPTION
**Pull request by DaceyTom2**
Description

Inbox items are derived from chatMessages and emails
emails can be selected based off of id selected
emails return full object with list of messages

TODO: 
teaser text and dayRead need to be generated dynamically
When a message is viewed the read indicator needs to be updated

### CODE REVIEW

File: src/store/modules/emails.js [hunk 1]
```diff
@@ -11,18 +10,16 @@ const getters = {
...
-      (sender) => sender.senderId === state.selectedSenderId
-    );
+    return getters.getMailObject.find((sender) => sender.id === id);
   },
   //returns all the emails, with most recent ones first
-  getEmailsOrderByDateAsc(state, getters) {
+  getMailObjectEmailsOrderByDateAsc: (state, getters) => (id) => {
     //get list of emails
-    const mail = getters.getSelectedSenderObject.emails;
+    var mailObject = { ...getters.getMailObjectById(id) };
```
Comment (nibivi77) [inline src/store/modules/emails.js:20]: Ewww a `var`! Get it out ! If only a `let` was here to save us!

File: src/store/modules/emails.js [hunk 2]
```diff
@@ -11,18 +10,16 @@ const getters = {
...
     //sorts a copy of the list of mail (getters should not mutate state)
-    const sortedMail = [...mail].sort((a, b) => {
+    mailObject.emails = [...mailObject.emails].sort((a, b) => {
```
Comment (nibivi77) [inline src/store/modules/emails.js]: Maybe we should have some error handling here just like the chatMessages.js?

What if mailObject is undefined?

File: src/store/modules/inbox.js [hunk 3]
```diff
@@ -2,34 +2,38 @@
...
+      });
+    });
+
+    rootGetters["emails/getMailObject"].forEach((item) => {
+      inboxItems.push({
+        id: item.id,
+        senderIcon: item.senderIcon,
+        senderName: item.senderName,
+        teaserText: "",
+        dayRead: "wed",
```
Comment (nibivi77) [inline src/store/modules/inbox.js:32]: If we can, we could call a function, pass the item, and return the date we need. Just an idea. (This can be implemented in a different PR)

P.S.: The same applies for teaserText.

**Review by nibivi77 — CHANGES_REQUESTED** [review]
Looking good Tom!

File: src/store/modules/emails.js (hunk 1 above)
Comment (DaceyTom2) [reply src/store/modules/emails.js]: It's gone... Pleassseee forgive me,I learned js so long ago

File: src/store/modules/inbox.js (hunk 3 above)
Comment (DaceyTom2) [reply src/store/modules/inbox.js:32]: Yes, I didn't want to jump into anything complex before establishing a baseline. Once we get this PR merged that can be done

File: src/store/modules/emails.js (hunk 2 above)
Comment (DaceyTom2) [reply src/store/modules/emails.js]: I must have added it in while you were commenting on this PR. However! I did add a check to make sure some emails existed like in chatMessages

File: src/store/modules/inbox.js (hunk 3 above)
Comment (nibivi77) [reply src/store/modules/inbox.js:32]: ok sounds good to me.

**Review by nibivi77 — APPROVED** [review]
This looks good to me!
